lib/agent.js

```diff
@@ -5,6 +5,8 @@
 
 'use strict'
 
+/* eslint sonarjs/cognitive-complexity: ["error", 23] -- TODO: https://issues.newrelic.com/browse/NEWRELIC-5252 */
```

Comment:
this should be on line 728 as _addIntrinsicAttrsFromTransaction is the function in violation. By doing this at the file level it sets cognitive complexity to error at 23 for this entire file

Reply:
ok we talked in slack. there's no nice way to override an eslint rule at a function level as it will apply for the remaining lines of the file. the only way to do this would be to override the value at start of function and restore after. since we plan on addressing these violations shortly after this PR lands, it is not worth the headache